src/components/text-editor/prosemirror-adapter/plugins/trigger-plugin.ts

```diff
@@ -0,0 +1,303 @@
+// import { ListItem } from 'src/components/list/list-item.types';
+
+import { Node, NodeType, Schema } from 'prosemirror-model';
+import { Plugin, PluginKey, EditorState } from 'prosemirror-state';
+import { ReplaceStep, ReplaceAroundStep } from 'prosemirror-transform';
+import { EditorView } from 'prosemirror-view';
+import { CustomElement } from 'src/interface';
+
+export const triggerPluginKey = new PluginKey('triggerPlugin');
+
+export type TriggerCharacter =
+    | '@'
+    | '#'
+    | '$'
+    | '!'
+    | '?'
+    | '&'
+    | '*'
+    | '%'
+    | '+'
+    | '-'
+    | '='
+    | '/'
+    | '\\'
+    | '^'
+    | '~'
+    | '`'
+    | ':'
+    | ';'
+    | '|'
+    | '.'
+    | ','
+    | '<'
+    | '>'
+    | '['
+    | ']'
+    | '{'
+    | '}'
+    | '('
+    | ')'
+    | "'";
+
+export type CustomElementInstance = Omit<CustomElement, 'attributes'> & {
+    attributes: Record<string, any>;
+};
+
+export type TextEditorNode = {
+    node: CustomElementInstance | string;
+    childs?: Array<TextEditorNode | string>;
+};
+
+export interface TextEditorInserter {
```

Comment:
Maybe this should just be named `TextEditor` if we want it to be able to add more methods to it in the future

```suggestion
export interface TextEditor {
```

Reply:
The name is maybe a bit generic?

Reply:
Fixed in latest commit

Reply:
Will this possibly become a bit confusing? We have the `text-editor` in lime-elements, and will have the `limebb-text-editor` as well (though perhaps this limebb component should be named differently).

Reply:
Yeah, I had the same thought